.github/workflows/python.yaml

```diff
@@ -136,6 +136,8 @@ jobs:
           pip install eva-decord
           pip install jj-pytorchvideo
           pip install qwen-vl-utils
+          pip install datamodel_code_generator
```

Comment:
Why does this model have this dependencies? Where can I refer to?

Reply:
2024-09-10 13:37:21,543 xinference.core.worker 688 ERROR    Failed to load model minicpm3-4b-1-0
Traceback (most recent call last):
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/worker.py", line 893, in launch_builtin_model
    await model_ref.load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
    return self._process_result_message(result)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
    result = await self._run_coro(message.message_id, coro)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
    return await coro
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/model.py", line 309, in load
    self._model.load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 755, in load
    super().load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 318, in load
    self._model, self._tokenizer = self._load_model(**kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 187, in _load_model
    tokenizer = AutoTokenizer.from_pretrained(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/models/auto/tokenization_auto.py", line 877, in from_pretrained
    tokenizer_class = get_class_from_dynamic_module(class_ref, pretrained_model_name_or_path, **kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 502, in get_class_from_dynamic_module
    final_module = get_cached_module_file(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 327, in get_cached_module_file
    modules_needed = check_imports(resolved_module_file)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 182, in check_imports
    raise ImportError(
ImportError: [address=0.0.0.0:45191, pid=708] This modeling file requires the following packages that were not found in your environment: datamodel_code_generator, jsonschema. Run `pip install datamodel_code_generator jsonschema`
2024-09-10 13:37:21,599 xinference.core.worker 688 ERROR    [request 13db8fd6-6f36-11ef-b9cb-0242c0a84902] Leave launch_builtin_model, error: [address=0.0.0.0:45191, pid=708] This modeling file requires the following packages that were not found in your environment: datamodel_code_generator, jsonschema. Run `pip install datamodel_code_generator jsonschema`, elapsed time: 289 s
Traceback (most recent call last):
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
    ret = await func(*args, **kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/worker.py", line 893, in launch_builtin_model
    await model_ref.load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
    return self._process_result_message(result)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
    result = await self._run_coro(message.message_id, coro)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
    return await coro
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/model.py", line 309, in load
    self._model.load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 755, in load
    super().load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 318, in load
    self._model, self._tokenizer = self._load_model(**kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 187, in _load_model
    tokenizer = AutoTokenizer.from_pretrained(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/models/auto/tokenization_auto.py", line 877, in from_pretrained
    tokenizer_class = get_class_from_dynamic_module(class_ref, pretrained_model_name_or_path, **kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 502, in get_class_from_dynamic_module
    final_module = get_cached_module_file(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 327, in get_cached_module_file
    modules_needed = check_imports(resolved_module_file)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 182, in check_imports
    raise ImportError(
ImportError: [address=0.0.0.0:45191, pid=708] This modeling file requires the following packages that were not found in your environment: datamodel_code_generator, jsonschema. Run `pip install datamodel_code_generator jsonschema`
2024-09-10 13:37:21,617 xinference.api.restful_api 559 ERROR    [address=0.0.0.0:45191, pid=708] This modeling file requires the following packages that were not found in your environment: datamodel_code_generator, jsonschema. Run `pip install datamodel_code_generator jsonschema`
Traceback (most recent call last):
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/api/restful_api.py", line 876, in launch_model
    model_uid = await (await self._get_supervisor_ref()).launch_builtin_model(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
    return self._process_result_message(result)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
    result = await self._run_coro(message.message_id, coro)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
    return await coro
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/supervisor.py", line 1026, in launch_builtin_model
    await _launch_model()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/supervisor.py", line 990, in _launch_model
    await _launch_one_model(rep_model_uid)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/supervisor.py", line 969, in _launch_one_model
    await worker_ref.launch_builtin_model(
  File "xoscar/core.pyx", line 284, in __pyx_actor_method_wrapper
    async with lock:
  File "xoscar/core.pyx", line 287, in xoscar.core.__pyx_actor_method_wrapper
    result = await result
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/utils.py", line 69, in wrapped
    ret = await func(*args, **kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/worker.py", line 893, in launch_builtin_model
    await model_ref.load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 231, in send
    return self._process_result_message(result)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/context.py", line 102, in _process_result_message
    raise message.as_instanceof_cause()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 656, in send
    result = await self._run_coro(message.message_id, coro)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/backends/pool.py", line 367, in _run_coro
    return await coro
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xoscar/api.py", line 384, in __on_receive__
    return await super().__on_receive__(message)  # type: ignore
  File "xoscar/core.pyx", line 558, in __on_receive__
    raise ex
  File "xoscar/core.pyx", line 520, in xoscar.core._BaseActor.__on_receive__
    async with self._lock:
  File "xoscar/core.pyx", line 521, in xoscar.core._BaseActor.__on_receive__
    with debug_async_timeout('actor_lock_timeout',
  File "xoscar/core.pyx", line 526, in xoscar.core._BaseActor.__on_receive__
    result = await result
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/core/model.py", line 309, in load
    self._model.load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 755, in load
    super().load()
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 318, in load
    self._model, self._tokenizer = self._load_model(**kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/xinference/model/llm/transformers/core.py", line 187, in _load_model
    tokenizer = AutoTokenizer.from_pretrained(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/models/auto/tokenization_auto.py", line 877, in from_pretrained
    tokenizer_class = get_class_from_dynamic_module(class_ref, pretrained_model_name_or_path, **kwargs)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 502, in get_class_from_dynamic_module
    final_module = get_cached_module_file(
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 327, in get_cached_module_file
    modules_needed = check_imports(resolved_module_file)
  File "/root/xinglin-data/xinf/lib/python3.10/site-packages/transformers/dynamic_module_utils.py", line 182, in check_imports
    raise ImportError(
ImportError: [address=0.0.0.0:45191, pid=708] This modeling file requires the following packages that were not found in your environment: datamodel_code_generator, jsonschema. Run `pip install datamodel_code_generator jsonschema`